docs/RFCS/20191009_gc_protected_timestamps.md

```diff
@@ -0,0 +1,597 @@
+state. In this way a range would always know whether it could GC some timestamp
+without consulting an external subsystem. There are however several downsides:
+
+ * Range now becomes even more bloated
+   * Range state is interacted with transactionally and thus it incurs even more
+     opportunity to race with other admin commands.
+ * Splits and merges gain complexity
+ * Clearing protected timestamps becomes transaction which touches all protected
+   ranges.
+
+Ignoring the cost of installing, clearing, and introspecting these protections,
+this initial straw man does not provide any safety mechanisms to ensure that
+protections expire or are removed. One could imagine that each protection
+installed in the ranges carries its own 
+   
+One could imagine a hybrid solution where we install both the (spans, ts) and
+some cluster-unique identifier for that protection and then there might exist
+some mechanism to invalidate protections. 
+
+### Data model
+
+There are levels to the alternatives considered. The most simplistic model
+considered was a single key which would store an encoded proto which roughly
+captures all of `protectedts.State`. This was deemed too simplistic and
+decidedly would not scale (though there's an interesting question about using
+column families to store the metadata and state all in one row).
+
+Another alternative was to keep a meta row and a list of protected timestamps
+but to further encode the protected timestamps as protos, i.e. encode
+`protectedtsbase.Record` and store it in a table. This would probably be totally
+fine.
+
+The SQL-heavy data model was primarily chosen out of a desire to utilize and
+familiarize myself with the data modeling which our users will be exposed to.
+
+I can certainly see arguments against the interleaved table.
+
+## Unresolved questions
+
+- Why not heartbeating or expiration?
+  
+Just using an expiration is much simpler. For jobs which will fail but not cause
+correctness problems in the case of undesired GC an expiration feels like a
+reasonable solution. The problem with expirations is how to choose a good one
+that isn't too lenient such that waiting for it to expire is too expensive but
+also isn't too aggressive such that jobs are actually at risk of hitting their
+expiration.
+
+For jobs which will cause correctness problems if undesired GC occurs then
+expirations on protections are almost as bad as no protections at all. At the
+very least there would need to be a mechanism to update the expiration. Once
+there's a mechanism to update an expiration I'm not sure how it differs from
+a heartbeat.
+
+Heartbeating is a problem for jobs which can be paused and resumed.
+
+- Will clients need to implement reconciliation?
+
+- How does the use of time-bound iterators interact with the efficiency of
+  this solution as written and more importantly the exact protection 
+  optimization?
+
+- Is it a bad idea to use SQL for the state as opposed to a pseudo-table like
```

Comment:
`system.rangelog` is prior-art for adding a SQL dependency to KV, so I'm less personally opposed to it than I was before.